src/App.tsx

```diff
@@ -1,222 +1,208 @@
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo, TodoTitle, SelectedType } from './types/Todo';
+import { ErrorType } from './components/ErrorType/ErrorType';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState<string>(SelectedType.ALL);
+  const [showAll, setShowAll] = useState(true);
+  const [isDeleting, setIsDeleting] = useState<number[]>([]);
+  const [isChangeAllTodos, setIsChangeAllTodos] = useState<number[]>([]);
+  const [isError, setIsError] = useState('');
   const user = useContext(AuthContext);
+  const [isAdding, setIsAdding] = useState(false);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [loaderVisibility, setIsLoaderVisibility] = useState<number>(0);
+
+  const handleError = (text: string) => {
+    setIsError(text);
+    setTimeout(() => {
+      setIsError('');
+    }, 5000);
+  };
+
+  const handleNewTodos = async () => {
+    if (!query.trim().length) {
+      handleError('Title can\'t be empty');
+
+      return;
+    }
+
+    setIsAdding(true);
+
+    try {
+      const newTodo = await createTodo(user?.id, query);
+
+      setIsLoaderVisibility(newTodo.id);
+
+      setTodos((prevState) => {
+        return [...prevState, newTodo];
+      });
+    } catch (error) {
+      handleError('Unable to add a todo');
+    } finally {
+      setIsAdding(false);
+      setTimeout(() => {
+        setIsLoaderVisibility(0);
+      }, 1000);
+    }
+  };
+
+  const removeTodo = useCallback(async (todoId: number) => {
+    setIsDeleting(prevIds => [...prevIds, todoId]);
+
+    try {
+      await deleteTodo(todoId);
+      setTodos(prevState => prevState.filter((todo) => todo.id !== todoId));
+    } catch (error) {
+      handleError('Unable to delete a todo');
+    }
+  }, []);
+
+  const changeTodo = useCallback(async (todoId: number, object: TodoTitle) => {
+    try {
+      const updatedTodo: Todo = await updateTodo(todoId, object);
+
+      setIsLoaderVisibility(updatedTodo.id);
+
+      setTodos(prevState => (prevState.map((a) => (a.id === todoId
+        ? updatedTodo
+        : a))
+      ));
+    } catch (error) {
+      handleError('Unable to update a todo');
+    } finally {
+      setTimeout(() => {
+        setIsLoaderVisibility(0);
+      }, 1000);
+    }
+  }, []);
+
+  const showAllTodos = () => {
+    todos.forEach(todo => {
+      changeTodo(todo.id, { completed: showAll });
+
+      setIsChangeAllTodos((prev) => [...prev, todo.id]);
+    });
+
+    setTimeout(() => {
+      setIsChangeAllTodos([]);
+    }, 1000);
+
+    setShowAll(!showAll);
+  };
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    getTodos(user?.id)
+      .then(res => setTodos(res))
+      .catch(() => setIsError);
```

Comment:
What did you want to express in the code with this entry? Can you explain please?
```suggestion
      .catch((error) => setIsError(error?.message));
```

Reply:
I recommend that you repeat [`try {...} catch {...}`](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Statements/try...catch)